### PR TITLE
Hibernate reactive - remove the Mutiny.Session bean

### LINKED ISSUE #30717: Hibernate reactive - deprecate the usage of injected Mutiny.Session
### Description

We should deprecate or even remove the `ReactiveSessionProducer` once we merge https://github.com/quarkusio/quarkus/pull/29761.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #30717

### CODE REVIEW

**Review by yrodiere — COMMENTED** [review]
Seems to make sense, but I can't comment on the specifics of Reactive, so I'll let Davide/Sanne do that.

File: integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java [hunk 1]
```diff
@@ -63,9 +60,8 @@ public Uni<String> reactiveRemoveTransientEntity() {
                     if (name == null)
                         throw new AssertionError("Database was not populated properly");
                 })
-                .chain(() -> mutinySession.merge(new GuineaPig(5, "Aloi")))
-                .chain(aloi -> mutinySession.remove(aloi))
-                .chain(() -> mutinySession.flush())
+                .chain(() -> sessionFactory
+                        .withTransaction(s -> s.merge(new GuineaPig(5, "Aloi")).chain(aloi -> s.remove(aloi))))
```
Comment (DavideD) [inline integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java:64]: I think you can change this with:
```
.chain(s::remove)
```

Comment (DavideD) [reply integration-tests/hibernate-reactive-postgresql/src/main/java/io/quarkus/it/hibernate/reactive/postgresql/HibernateReactiveTestEndpoint.java:64]: The same is true in other places

File: integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2AlternativeTestEndpoint.java [hunk 2]
```diff
@@ -75,15 +72,14 @@ public Uni<String> reactiveRemoveManagedEntity() {
...
-                .onItem().transform(pig -> {
-                    if (NEW_NAME.equals(pig.getName())) {
-                        throw new AssertionError("Pig already had name " + NEW_NAME);
-                    }
-                    pig.setName(NEW_NAME);
-                    return pig;
-                })
-                .chain(() -> mutinySession.flush())
+                .chain(() -> sessionFactory.withTransaction(s -> s.find(GuineaPig.class, 5)
+                        .onItem().transform(pig -> {
```
Comment (DavideD) [inline integration-tests/hibernate-reactive-db2/src/main/java/io/quarkus/it/hibernate/reactive/db2/HibernateReactiveDB2AlternativeTestEndpoint.java:76]: You could change this to:
```
.invoke(pig -> {
   // same code without return pig
});
```

**Review by DavideD — APPROVED** [review]
LGTM

There are several places where we could clean up the code, but it's not that important

**Comment by mkouba** [comment]
> There are several places where we could clean up the code, but it's not that important

For sure we could. I've just tried to adapt the original code...

**Comment by mkouba** [comment]
For the record - I've removed the producer method because there's no easy way to deprecate a bean. I mean we could detect all `Mutiny.Session` injection points and log a warning during build but a user gets no warning in the IDE etc. So I think it's better to remove the bean completely.

**Comment by Sanne** [comment]
> For the record - I've removed the producer method because there's no easy way to deprecate a bean. I mean we could detect all `Mutiny.Session` injection points and log a warning during build but a user gets no warning in the IDE etc. So I think it's better to remove the bean completely.

Makes sense! Many thanks

**Review by Sanne — APPROVED** [review]
Great to remove this!

**Pull request merged**